sdk/src/main/java/co/omise/android/ui/AtomeFormFragment.kt

```diff
@@ -0,0 +1,254 @@
+    }
+
+    private fun isEmailValid(emailEdit: OmiseEditText): Boolean {
+        return android.util.Patterns.EMAIL_ADDRESS.matcher(emailEdit.text!!).matches()
+    }
+
+    private fun isPhoneNumberValid(phoneNumberEdit: OmiseEditText): Boolean {
+        return android.util.Patterns.PHONE.matcher(phoneNumberEdit.text!!).matches()
+    }
+
+    private fun isCountryCodeValid(countryCodeEdit: OmiseEditText): Boolean {
+        return countryCodeEdit.length() == 2
+    }
+
+    private fun onBillingShippingCheckboxClicked(view: View) {
+        if (view is CheckBox) {
+            val checked: Boolean = view.isChecked
+
+            when (view.id) {
+                R.id.checkbox_billing_shipping -> {
+                    if (checked) {
+                        billing_address.visibility = GONE
+                    } else {
+                        billing_address.visibility = VISIBLE
+                    }
+                }
+            }
+        }
+    }
+
+    private fun updateSubmitButton() {
+        submitButton.isEnabled = (emailEdit.text?.isEmpty() == true || isEmailValid(emailEdit)) &&
+                isPhoneNumberValid(phoneNumberEdit) &&
+                shippingStreetEdit.isValid &&
+                shippingPostalEdit.isValid &&
+                shippingCityEdit.isValid &&
+                isCountryCodeValid(shippingCountryEdit) &&
+                (billingCityEdit.text?.isEmpty() == true || isCountryCodeValid(billingCountryEdit))
+    }
+
+    private fun submitForm() {
+        val requester = requester ?: return
+        val requestBuilder = Source.CreateSourceRequestBuilder(requester.amount, requester.currency, SourceType.Atome)
+
+        val fullName = fullNameEdit.text?.toString()?.trim().orEmpty()
+        requestBuilder.name(fullName)
+
+        val email = emailEdit.text?.toString()?.trim().orEmpty()
+        requestBuilder.email(email)
+
+        val phoneNumber = phoneNumberEdit.text?.toString()?.trim().orEmpty()
+        requestBuilder.phoneNumber(phoneNumber)
+
+        val shippingStreet = shippingStreetEdit.text?.toString()?.trim().orEmpty()
+        val shippingPostal = shippingPostalEdit.text?.toString()?.trim().orEmpty()
+        val shippingCity = shippingCityEdit.text?.toString()?.trim().orEmpty()
+        val shippingCountry = shippingCountryEdit.text?.toString()?.trim().orEmpty()
+        requestBuilder.shipping(
+            Shipping(
+                street1 = shippingStreet,
+                postalCode = shippingPostal,
+                city = shippingCity,
+                country = shippingCountry
+            )
+        ).items(
+            listOf(
+                Item(
+                    "3427842",
+                    "Shoes",
+                    "Prada shoes",
+                    "1",
+                    requester.amount.toString(),
+                    "www.kan.com/product/shoes",
+                    "www.kan.com/product/shoes/image",
```

Comment:
hahahahahaha 😆

Reply:
Example values ​​you wrote Confluence is something that matches this 😆